components/dialog/dialog-mixin.js

```diff
@@ -124,7 +124,12 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 	}
 
 	resize() {
-		return this._updateSize();
+		return new Promise(resolve => {
+			setTimeout(async() => {
+				await this._updateSize();
+				resolve();
+			});
+		});
```

Comment:
This resolves a separate issue I found both in Chrome and Firefox without my changes. The list item in the demo page doesn't finish rendering by the time the async dialog calls this resize method, hence the dialog sizes to zero height. It's a similar sort of problem and using rAF does resolve it in Chrome, but much like the other issue, we need `setTimeout(, 0)` to resolve this in Firefox.